NDTensors/src/lib/Sectors/test/test_category_product.jl

```diff
@@ -50,7 +60,7 @@ using Test: @test, @testset, @test_throws
     q22 = (N=U1(2),) × (J=SU2(2),)
 
     @test q1h ⊗ q1h == q20 ⊕ q21
-    @test q10 ⊗ q1h == [q2h]
+    @test q10 ⊗ q1h == gradedrange([q2h => 1])
```

Comment:
I think it would be good to discuss what this case should output.

Reply:
As mentioned a `CategoryProduct` should act more like an `AbstractCategory`, since `U(1) × SU(2)` is a non-abelian group I think this is the best option. The alternative would be to return intermediate steps that would look like a Cartesian product of category-specific ranges.

Note that the test currently fails as the implementation of `⊗` is still a work in progress for  `CategoryProduct`

Reply:
That's a good point, I missed that this was a `CategoryProduct`, agreed that it makes sense to choose a consistent output for any `CategoryProduct` (which covers both the abelian and non-abelian cases), in which case `GradedUnitRange` is the clear choice.